qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/QuteLanguageServer.java

```diff
@@ -288,4 +293,14 @@ public void setTrace(SetTraceParams params) {
 		// to avoid having error in vscode, the method is implemented
 		// FIXME : implement the behavior of this method.
 	}
+
+	@Override
+	public Template getTemplate(String uri) {
+		TextDocumentIdentifier textDocument = new TextDocumentIdentifier(uri);
+		TemplateFileTextDocumentService templateTextDocumentService = textDocumentService.getTemplateTextDocumentService(textDocument);

```

Comment:
Define  a FileType enum like this:

```java
package com.redhat.qute.ls;

public enum FileType {

	TEMPLATE,
	JAVA;
}
```

In AbstractTextDocumentService, defines an abstract method getFileType() like this:

`public abstract FileType getFileType();`

Implement getFileType for the both implementation of AbstractTextDocumentService.

Replace your method like this:

```java
@Override
public Template getTemplate(String uri) {
	TextDocumentIdentifier textDocument = new TextDocumentIdentifier(uri);
	AbstractTextDocumentService textDocumentService = textDocumentService.getTextDocumentService(textDocument);
	if (textDocumentService.getFileType() = FileType.TEMPLATE) {
		TemplateFileTextDocumentService templateTextDocumentService = (TemplateFileTextDocumentService) textDocumentService;
		QuteTextDocument document = templateTextDocumentService.getDocument(uri);
		return document != null ? document.getModel() : null;
	}
	return null;
}
```